iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ConfiguredBusImpl.java

```diff
@@ -71,7 +71,40 @@ public int getConnectedTerminalCount() {
 
     @Override
     public TerminalExt getTerminalReference() {
-        return getConnectedTerminalStream().findFirst().orElseGet(() -> getTerminals().isEmpty() ? null : getTerminals().get(0));
+        return getConnectedTerminalStream().findFirst().orElseGet(() -> {
+            if (voltageLevel.getTopologyKind() == TopologyKind.BUS_BREAKER) {
+                // in bus-breaker voltage level, the terminal reference can be a connected terminal in another bus linked to this one by one or several closed switches
+                VoltageLevel.BusBreakerView view = voltageLevel.getBusBreakerView();
+                return view.getBusStream()
+                        .map(b -> getTerminalReference(b, this, view))
+                        .filter(Objects::nonNull)
+                        .findFirst()
+                        .orElse(null);
+            }
+            return null;
+        });
+    }
+
+    private static TerminalExt getTerminalReference(Bus b1, Bus b2, VoltageLevel.BusBreakerView view) {
```

Comment:
you wrote in comment above that there could be several closed switches between the two buses, but here I'm not so sure... it looks like you're only checking if they're directly connected?

Reply:
I'm closing, I was mistaken! it's perfect.